simulations/threaded_simulation.py

```diff
@@ -0,0 +1,55 @@
+from time import sleep
+
+from simulations.validator_client import ValidatorClient
+
+
+class ThreadedSimulation:
+    def __init__(
+            self,
+            validator_set,
+            msg_strategy,
+            protocol,
+            network,
+            total_rounds,
+            report_interval,
+            display=False,
+            save=True
+    ):
+        self.length_in_seconds = total_rounds
+        self.report_seconds = report_interval
+        self.network = network
+        self.validator_set = validator_set
+
+        self.validator_clients = [
+            ValidatorClient(validator, network) for validator in validator_set
+        ]
+        for validator_client in self.validator_clients:
+            validator_client.should_make_new_message = msg_strategy
+
+        self.plot_tool = protocol.PlotTool(False, save, network.global_view, validator_set)
+
+    def run(self):
+        print("starting simulation")
+        self._send_initial_messages()
+
+        for validator_client in self.validator_clients:
+            validator_client.start()
```

Comment:
This is looking very dope!

Might be helpful to have a comment here that we are initializing threads and `start()` is apart of the Thread Class. Not  exactly sure how validator_client.run() gets hit by this but I see it is working :)